`LocalFinalVariableName` must include `_`

I have read check documentation: https://checkstyle.sourceforge.io/checks/naming/localfinalvariablename.html#LocalFinalVariableName
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="LocalFinalVariableName"/>
  </module>
</module>

/var/tmp $ cat Test.java
public class Test {
	public void foo() {
		try (var _ = lock()) {
		} catch (Exception exception) {
		}
	}

	public AutoCloseable lock() {
		return null;
	}
}

/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-X.XX-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /home/boris/Test.java:3:26: Name '_' must match pattern '^[a-z][a-zA-Z0-9]*$'. [LocalFinalVariableName]
Audit done.
Checkstyle ends with 1 errors.
```

`_` is a valid identifier in Java 22 and should be supported by Checkstyle. I fixed it locally by using `<property name="format" value="^(_|[a-z][a-zA-Z0-9]*)$"/>`.

--- a/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java
+++ b/src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheck.java
@@ -34,7 +34,7 @@ import com.puppycrawl.tools.checkstyle.utils.ScopeUtil;
  * <li>
  * Property {@code format} - Sets the pattern to match valid identifiers.
  * Type is {@code java.util.regex.Pattern}.
- * Default value is {@code "^[a-z][a-zA-Z0-9]*$"}.
+ * Default value is {@code "^([a-z][a-zA-Z0-9]*|_)$"}.
  * </li>
  * <li>
  * Property {@code tokens} - tokens to check
@@ -68,7 +68,7 @@ public class LocalFinalVariableNameCheck
 
     /** Creates a new {@code LocalFinalVariableNameCheck} instance. */
     public LocalFinalVariableNameCheck() {
-        super("^[a-z][a-zA-Z0-9]*$");
+        super("^([a-z][a-zA-Z0-9]*|_)$");
     }
 
     @Override

--- a/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/naming/LocalFinalVariableNameCheck.xml
+++ b/src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/naming/LocalFinalVariableNameCheck.xml
@@ -10,7 +10,7 @@
  are considered to be a local, final variables.
  &lt;/p&gt;</description>
          <properties>
-            <property default-value="^[a-z][a-zA-Z0-9]*$"
+            <property default-value="^([a-z][a-zA-Z0-9]*|_)$"
                       name="format"
                       type="java.util.regex.Pattern">
                <description>Sets the pattern to match valid identifiers.</description>

--- a/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheckTest.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/checks/naming/LocalFinalVariableNameCheckTest.java
@@ -50,7 +50,7 @@ public class LocalFinalVariableNameCheckTest
     public void testDefault()
             throws Exception {
 
-        final String pattern = "^[a-z][a-zA-Z0-9]*$";
+        final String pattern = "^([a-z][a-zA-Z0-9]*|_)$";
 
         final String[] expected = {
             "126:19: " + getCheckMessage(MSG_INVALID_PATTERN, "CDE", pattern),
@@ -117,6 +117,23 @@ public class LocalFinalVariableNameCheckTest
         final String[] expected = CommonUtil.EMPTY_STRING_ARRAY;
         verifyWithInlineConfigParser(
                 getPath("InputLocalFinalVariableNameTryResourcesJava9.java"), expected);
+    }
+
+    @Test
+    public void testUnnamedVariables() throws Exception {
+        final String pattern = "^([a-z][a-zA-Z0-9]*|_)$";
+
+        final String[] expected = {
+            "18:18: " + getCheckMessage(MSG_INVALID_PATTERN, "__", pattern),
+            "21:32: " + getCheckMessage(MSG_INVALID_PATTERN, "__", pattern),
+            "29:24: " + getCheckMessage(MSG_INVALID_PATTERN, "__", pattern),
+            "31:24: " + getCheckMessage(MSG_INVALID_PATTERN, "_BAD", pattern),
+            "37:19: " + getCheckMessage(MSG_INVALID_PATTERN, "__", pattern),
+            "38:19: " + getCheckMessage(MSG_INVALID_PATTERN, "_BAD", pattern),
+        };
+        verifyWithInlineConfigParser(
+                getNonCompilablePath("InputLocalFinalVariableNameUnnamedVariables.java"),
+                expected);
     }
 
 }

--- a/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/localfinalvariablename/InputLocalFinalVariableNameUnnamedVariables.java
+++ b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/localfinalvariablename/InputLocalFinalVariableNameUnnamedVariables.java
@@ -1,0 +1,44 @@
+/*
+LocalFinalVariableName
+
+*/
+
+//non-compiled with javac: Compilable with Java21
+package com.puppycrawl.tools.checkstyle.checks.naming.localfinalvariablename;
+
+public class InputLocalFinalVariableNameUnnamedVariables {
+
+    void testTryWithResource(Object obj) {
+        try (var _ = lock()) {
+
+        }
+        catch (final Exception _) {
+        }
+
+        try (var __ = lock()) {   // violation, 'Name '__' must match pattern*.'
+
+        }
+        catch (final Exception __) {  // violation, 'Name '__' must match pattern*.'
+        }
+
+    }
+
+    void testEnhancedForLoop() {
+        for (final var _ : new int[0]) {
+        }
+        for (final var __ : new int[0]) { // violation, 'Name '__' must match pattern*.'
+        }
+        for (final var _BAD : new int[0]) { // violation, 'Name '_BAD' must match pattern*.'
+        }
+    }
+
+    void testLocalVariable(Object obj) {
+        final var _ = obj;
+        final var __ = obj; // violation, 'Name '__' must match pattern*.'
+        final var _BAD = obj; // violation, 'Name '_BAD' must match pattern*.'
+    }
+
+    public AutoCloseable lock() {
+        return null;
+    }
+}

--- a/src/xdocs/checks/naming/localfinalvariablename.xml
+++ b/src/xdocs/checks/naming/localfinalvariablename.xml
@@ -30,7 +30,7 @@
               <td>format</td>
               <td>Sets the pattern to match valid identifiers.</td>
               <td><a href="../../property_types.html#Pattern">Pattern</a></td>
-              <td><code>&quot;^[a-z][a-zA-Z0-9]*$&quot;</code></td>
+              <td><code>&quot;^([a-z][a-zA-Z0-9]*|_)$&quot;</code></td>
               <td>3.0</td>
             </tr>
             <tr>

